pysite/views/main/jams/retract.py

```diff
@@ -0,0 +1,84 @@
+from werkzeug.exceptions import BadRequest
+
+from pysite.base_route import RouteView
+from pysite.decorators import csrf
+from pysite.mixins import DBMixin, OAuthMixin
+
+BANNABLE_STATES = ["preparing", "running"]
```

Comment:
do we need to modify this? if not, why not make it a tuple?

Reply:
Done.